src/components/application_manager/test/commands/mobile/add_command_request_test.cc

```diff
@@ -0,0 +1,195 @@
+using am::ApplicationManager;
+using am::commands::MessageSharedPtr;
+using am::ApplicationSharedPtr;
+using am::MockMessageHelper;
+using ::testing::_;
+using ::utils::SharedPtr;
+using ::testing::Return;
+
+
+using ::test::components::application_manager_test::MockApplication;
+using usage_statistics_test::MockStatisticsManager;
+using am::event_engine::EventObserver;
+
+
+namespace smart_objects = NsSmartDeviceLink::NsSmartObjects;
+namespace custom_str = utils::custom_string;
+namespace strings = ::application_manager::strings;
+namespace hmi_response = ::application_manager::hmi_response;
+
+class AddCommandRequestImplTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+};
+
+const hmi_apis::FunctionID::eType kInvalidFunctionId_ =
+    hmi_apis::FunctionID::INVALID_ENUM;
+
+TEST_F(AddCommandRequestImplTest,  GetRunMethods_SUCCESS){
+//#ifdef __UNIT_TEST__
+//     friend AddCommandRequest;
+//   #endif
+     MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+
+     smart_objects::SmartObject menu_params = smart_objects::SmartObject(smart_objects::SmartType_Map);
+     menu_params[am::hmi_request::parent_id] = 1;
+     menu_params[strings::position] = 10;
+     menu_params[strings::menu_name] = "LG";
+
+
+     smart_objects::SmartObject msg_params = smart_objects::SmartObject(smart_objects::SmartType_Map);
+     msg_params[strings::cmd_id] = 1;
+     msg_params[strings::menu_params] = menu_params;
+     msg_params[strings::app_id] = 34;
+    (*msg)[strings::msg_params] = msg_params;
+
+
+     using namespace application_manager::commands;
+     utils::SharedPtr<AddCommandRequest> req = CreateCommand<AddCommandRequest>(msg);
+
+     MockAppPtr mock_app = CreateMockApp();
+     ON_CALL(app_mngr_, application(_)).WillByDefault(Return(mock_app));
+     EXPECT_CALL(*mock_app, FindCommand(_));
+     EXPECT_CALL(*mock_app, AddCommand(_, _));
+
+     MockMessageHelper* mock_message_helper = MockMessageHelper::message_helper_mock();
+
+     EXPECT_CALL(*mock_message_helper,
+                  VerifyImage(_, _, _));
+
+      req->Run();
+}
+
+TEST_F(AddCommandRequestImplTest, OnEvent_SUCCESS) {
+
+    MessageSharedPtr msg1 = CreateMessage(smart_objects::SmartType_Map);
+    smart_objects::SmartObject menu_params1 = smart_objects::SmartObject(smart_objects::SmartType_Map);
+    menu_params1[am::hmi_request::parent_id] = 1;
+    menu_params1[strings::position] = 10;
+    menu_params1[strings::menu_name] = "LG";
+    smart_objects::SmartObject msg_params1 = smart_objects::SmartObject(smart_objects::SmartType_Map);
+    msg_params1[strings::cmd_id] = 1;
+    msg_params1[strings::menu_params] = menu_params1;
+    msg_params1[strings::app_id] = 34;
+   (*msg1)[strings::msg_params] = msg_params1;
+    using namespace application_manager::commands;
+    utils::SharedPtr<AddCommandRequest> req1 = CreateCommand<AddCommandRequest>(msg1);
+
+ /*   MessageSharedPtr msg2 = CreateMessage(smart_objects::SmartType_Map);
```

Comment:
@wolfylambova22, why do you comment this code? If it's redundant, please, remove it.